src/main/java/frc/robot/wrappers/NeoSparkMaxPid.java

```diff
@@ -0,0 +1,143 @@
+package frc.robot.wrappers;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxRelativeEncoder;
+import com.revrobotics.CANSparkMax.ControlType;
+
+public class NeoSparkMaxPid {
+
+	private final CANSparkMax motor;
+
+	private final RelativeEncoder encoder;
+
+	private static final int HALL_COUNTS_PER_REV = 42;
+
+	/**
+	 * Constructor for Neo motor controlled by Spark Max
+	 * controller wrapping up PIDF algorithms. Defaults
+	 * to using 1 tick per revolution (almost certainly
+	 * not the gear ratio of the actual motor). Tune the
+	 * constant parameters to achieve desired outputs in
+	 * terms of speed and accuracy.
+	 * @param canId HardwareMapped ID
+	 * @param p	 PIDF Position constant
+	 * @param i	 PIDF Integral constant
+	 * @param d	 PIDF Derivative constant
+	 * @param f	 PIDF Feed Forward constant
+	 */
+	public NeoSparkMaxPid(final int canId,  final double p,
+											final double i,
+											final double d,
+											final double f) {
+		motor = new CANSparkMax(canId,
+						CANSparkMax.MotorType.kBrushless);
+
+		motor.getPIDController().setP(p);
+		motor.getPIDController().setI(i);
+		motor.getPIDController().setD(d);
+		motor.getPIDController().setFF(f);
+		motor.getPIDController().setIZone(0);
+		motor.getPIDController().setOutputRange(-1, 1);
+
+		encoder = motor.getEncoder(SparkMaxRelativeEncoder.Type.kHallSensor, HALL_COUNTS_PER_REV);
```

Comment:
Does the basic getEncoder() not work here?

Reply:
After closer inspection of the API, we believe it should, fixed